### ISSUE TITLE
Cannot read property 'callHook' of undefined

### ISSUE THREAD

**Issue by dmitriy-grablyov** [nuxt/bridge#856]
### Environment

```
- Operating System: Darwin
- Node Version:     v16.20.0
- Nuxt Version:     2.17.0
- Nitro Version:    2.5.2
- Package Manager:  npm@8.19.4
- Builder:          vite
- User Config:      bridge, server, head, plugins, serverHandlers, devServerHandlers, devServer, typescript, nitro, buildModules
- Runtime Modules:  -
- Build Modules:    (), @nuxt/bridge@3.0.0-28142286.528efb7
```

### Reproduction

https://github.com/dmitriy-grablyov/nuxt-bridge-callhook-bug

Please follow the readme

### Describe the bug

I am aware there have been at least two reports related to this issue:
- https://github.com/nuxt/bridge/issues/314
- https://github.com/nuxt/bridge/issues/138

However it seems both have been closed with 'user configuration error' resolution.
I trid now to create the most minimalistic example (see steps to reproduce), nothing extra, no components, no store/middlewares or anything like, just nuxt, nuxt bridge and a page with a simple 'Hello world'.

### Additional context

As mentioned in my comment in https://github.com/nuxt/bridge/issues/314

I tried to trace an error and only was able to trace it to this line https://github.com/nuxt/bridge/blob/1313524c5c84421837263d45202a64229881f864/packages/bridge/src/runtime/app.ts#L39
In runtime `ctx.$_nuxtApp` evaluates to undefined and I tried hard, but was not able to find the place in the sources where  `ctx.$_nuxtApp` is getting actually assigned to context. Where does this magic happen?

Thanks in advance!

### Logs

```shell
% npm run start    

> nuxt-bug@1.0.0 start
> nuxi start

Nuxt CLI v3.0.0-27386759.b449d0b                                                                                                                                                18:03:55
ℹ Node.js version: 16.20.0                                                                                                                                                      18:03:55
ℹ Preset: server                                                                                                                                                                18:03:55
ℹ Working dir: .output                                                                                                                                                          18:03:55
ℹ Starting preview command: node ./server/index.mjs                                                                                                                             18:03:55
                                                                                                                                                                                18:03:55
Listening on http://localhost:3000/
Cannot read properties of undefined (reading 'callHook')
  at file://./server/chunks/app/server.mjs:11831:11  
  at file://./server/chunks/app/server.mjs:11812:5  
  at createApp (file://./server/chunks/app/server.mjs:11963:11)  
  at async server (file://./server/chunks/app/server.mjs:12049:27)  
  at async renderToString (file://./server/chunks/app/render.mjs:258:19)  
  at async renderMiddleware (file://./server/chunks/app/render.mjs:572:20)  
  at async handle (file://./server/node_modules/h3/dist/index.mjs:601:19)
Cannot read properties of undefined (reading 'callHook')
  at file://./server/chunks/app/server.mjs:11831:11  
  at file://./server/chunks/app/server.mjs:11812:5  
  at createApp (file://./server/chunks/app/server.mjs:11963:11)  
  at processTicksAndRejections (node:internal/process/task_queues:96:5)  
  at async server (file://./server/chunks/app/server.mjs:12049:27)  
  at async renderToString (file://./server/chunks/app/render.mjs:258:19)  
  at async renderMiddleware (file://./server/chunks/app/render.mjs:572:20)  
  at async handle (file://./server/node_modules/h3/dist/index.mjs:601:19)
```

**Comment by wattanx** [comment]
@dmitriy-grablyov 
It looks like the version of Nuxt Bridge is out of date. Does it still occur with the latest version? (`3.0.0-28142286.528efb7`)

**Comment by dmitriy-grablyov** [comment]
Hi @wattanx , thanks for the reply. Unfortunately the problem still occurs, I've updated test project accordingly.

```
[nuxt] [request error] [unhandled] [500] Cannot read properties of undefined (reading 'callHook')
  at ./server/chunks/app/server.mjs:6259:11  
  at callWithNuxt (./server/chunks/app/server.mjs:6229:20)  
  at ./server/chunks/app/server.mjs:6234:26  
  at createApp (./server/chunks/app/server.mjs:6401:11)  
  at async server (./server/chunks/app/server.mjs:6490:27)  
  at async Object.renderToString (./server/chunks/handlers/renderer.mjs:217:19)  
  at async ./server/chunks/handlers/renderer.mjs:612:21  
  at async ./server/chunks/handlers/renderer.mjs:22:22  
  at async Object.handler (./server/node_modules/h3/dist/index.mjs:1284:19)  
  at async Server.toNodeHandle (./server/node_modules/h3/dist/index.mjs:1359:7)
```

**Comment by wattanx** [comment]
Perhaps the command is wrong.
If `nitro: true` and `vite: true`, you need to use the `nuxi preview` command.
(The `nuxi start` does not seem to exist.)
https://nuxt.com/docs/api/commands/preview

**Comment by dmitriy-grablyov** [comment]
I changed start to preview, to no avail.
I see that your project works, however it is a completely different setup, with typescript vs no-typescript, yarn vs npm and different bridge configuration in nuxt config file.

Would you be able to try to clone my repository and run my project in your environment?

I have one another suspicion, that it might be related to apple M1 . It should be highly unlikely the case, but I still want to check.

**Comment by wattanx** [comment]
I have examined the contents of your repositories and it appears that the `capi` setting is incorrect.
With `capi: false` I got the error you mentioned.
https://stackblitz.com/edit/github-3fgb91-sixttg

If you set the following, the error will not occur.
```ts
capi: {
  legacy: false,
},
```
I see that this one is listed in the documentation.
https://nuxt.com/docs/bridge/overview#feature-flags

**Comment by dmitriy-grablyov** [comment]
oh, well, thanks, I tried that and it works!

On the other hand, to be honest, I don't know, what I've just changed, it looks like the documentation lacks an explanation for these flags.  Especially that `capi: false` is listed as a default setting. 
What is actually the difference between `capi: false` and `capi: { legacy: false }`?

And although it is required to use nuxt@2.17.0 for upgrade, which comes with newer version of vue, we in our project did not explicitly use composition api. Nor it is mentioned anywhere in my test repo. So the nature of an error is unclear.

In any case thanks for your support.
